mm2src/coins/nft_storage/sql_storage.rs

```diff
@@ -0,0 +1,885 @@
+            let sql_string = sql_builder.sql()?.trim_end_matches(';').to_string();
+            Ok(sql_string)
+        })
+        .collect();
+
+    let union_sql_strings = union_sql_strings?;
+    let union_sql = union_sql_strings.join(" UNION ALL ");
+    let mut final_sql_builder = SqlQuery::select_from_union_alias(conn, union_sql.as_str(), "nft_list")?;
+    final_sql_builder.order_desc("nft_list.block_number")?;
+    drop_mutability!(final_sql_builder);
+    Ok(final_sql_builder)
+}
+
+fn get_nft_tx_builder_preimage(
+    conn: &Connection,
+    chains: Vec<Chain>,
+    filters: Option<NftTxHistoryFilters>,
+) -> MmResult<SqlQuery, SqlError> {
+    let union_sql_strings: MmResult<Vec<_>, SqlError> = chains
+        .into_iter()
+        .map(|chain| {
+            let table_name = nft_tx_history_table_name(&chain);
+            validate_table_name(&table_name)?;
+            let sql_builder = nft_history_table_builder_preimage(conn, table_name.as_str(), filters.clone())?;
+            let sql_string = sql_builder.sql()?.trim_end_matches(';').to_string();
+            Ok(sql_string)
+        })
+        .collect();
+
+    let union_sql_strings = union_sql_strings?;
+    let union_sql = union_sql_strings.join(" UNION ALL ");
+    let mut final_sql_builder = SqlQuery::select_from_union_alias(conn, union_sql.as_str(), "nft_history")?;
+    final_sql_builder.order_desc("nft_history.block_timestamp")?;
+    drop_mutability!(final_sql_builder);
+    Ok(final_sql_builder)
+}
+
+fn nft_history_table_builder_preimage<'a>(
+    conn: &'a Connection,
+    table_name: &'a str,
+    filters: Option<NftTxHistoryFilters>,
+) -> Result<SqlQuery<'a>, SqlError> {
+    let mut sql_builder = SqlQuery::select_from(conn, table_name)?;
+    if let Some(filters) = filters {
+        if filters.send && !filters.receive {
+            sql_builder.sql_builder().and_where_eq("status", "'Send'");
+        } else if filters.receive && !filters.send {
+            sql_builder.sql_builder().and_where_eq("status", "'Receive'");
+        }
+
+        if filters.from_date.is_some() && filters.to_date.is_some() {
+            sql_builder.sql_builder().and_where(format!(
+                "block_timestamp BETWEEN '{}' and '{}'",
+                filters.from_date.unwrap(),
+                filters.to_date.unwrap()
+            ));
+        } else if filters.from_date.is_some() {
+            sql_builder
+                .sql_builder()
+                .and_where(format!("block_timestamp >= '{}'", filters.from_date.unwrap()));
+        } else if filters.to_date.is_some() {
+            sql_builder
+                .sql_builder()
+                .and_where(format!("block_timestamp <= '{}'", filters.to_date.unwrap()));
+        }
+    }
+    drop_mutability!(sql_builder);
+    Ok(sql_builder)
+}
+
+fn nft_table_builder_preimage<'a>(conn: &'a Connection, table_name: &'a str) -> MmResult<SqlQuery<'a>, SqlError> {
```

Comment:
Why is this needed when it's used only once and `SqlQuery::select_from` is used a lot instead of it in multiple places?

Reply:
hmm, I believe it was a part of my "legacy" code, where I was playing with SqlQuery first time.
I removed it, thanks!